pkgs/development/python-modules/asgi-lifespan/default.nix

```
@@ -0,0 +1,61 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+
+# build-system
+, setuptools-scm
+
+# dependencies
+, twine
+, wheel
+, attrs
+, autoflake
+, black
+, mypy
+, pytest
+, ruff
+, starlette
+, trio
+, httpx
+, setuptools
+
+# tests
+ }:
+
+buildPythonPackage rec {
+  pname = "asgi-lifespan";
+  version = "2.1.0";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-Xi7/rwv+OYKc8tZOfsxHx9htZ2plmfevujeMMfXjowg=";
+  };
+
+  postPatch = ''
+    # don't test bash builtins
+    #rm testing/test_argcomplete.py
+  '';
+
+  nativeBuildInputs = [
+    setuptools-scm
+  ];
+
+  propagatedBuildInputs = [
+    setuptools
+    starlette
+  ];
+
+  nativeCheckInputs = [
+  ];
+
+  doCheck = false;
+
+  meta = with lib; {
+    changelog = "https://github.com/florimondmanca/asgi-lifespan/releases/tag//${version}";
+    description = "Programmatically send startup/shutdown lifespan events into ASGI applications.";
+    homepage = "https://github.com/florimondmanca/asgi-lifespan/releases/tag";
```

This URL gives a 404.

```suggestion
    homepage = "https://github.com/florimondmanca/asgi-lifespan/";
```